packages/main/src/Tokenizer.ts

```diff
@@ -107,53 +150,126 @@ enum ClipboardDataOperation {
 	],
 })
 
-@event("token-delete", {
+/**
+ * Fired when a token is deleted (delete icon, delete or bacspace is pressed)
+ *
+ * @param {HTMLElement} ref DOM ref of the token to be deleted.
+ * @public
+ */
+@event<TokenizerTokenDeleteEventDetail>("token-delete", {
 	detail: {
 		ref: { type: HTMLElement },
 	},
 })
 
-@event("show-more-items-press", {
+/**
+ * Fired when token selection is changed by user interaction
+ *
+ * @param {Array<Token>} selectedTokens An array of the selected items.
+ * @public
+ */
+@event<TokenizerSelectionChangeEventDetail>("selection-change", {
 	detail: {
-		ref: { type: HTMLElement },
+		selectedTokens: { type: Array },
+	},
+})
+
+/**
+ * Fired when a dialog button is pressed.
+ *
+ * @param {Boolean} confirm Indicates if the action is confirm.
+ * @public
+ */
+@event<TokenizerDialogButtonPressDetail>("dialog-button-press", {
+	detail: {
+		confirm: { type: Boolean },
 	},
 })
 
+/**
+ * Fired when nMore link is pressed.
+ * @private
+ */
+@event("show-more-items-press")
+
+/**
+ * Fired before nMore Popover is opened.
+ * @private
+ */
 @event("before-more-popover-open")
-class Tokenizer extends UI5Element {
-	@property({ type: Boolean })
-	showMore!: boolean;
 
+class Tokenizer extends UI5Element {
+	/**
+	 * Defines whether the component is disabled.
+	 *
+	 * **Note:** A disabled component is completely noninteractive.
+	 * @default false
+	 * @public
+	 */
 	@property({ type: Boolean })
 	disabled!: boolean;
 
 	/**
-	 * Prevent opening of n-more Popover when label is clicked
-	 * @private
+	 * Defines whether the component is read-only.
+	 *
+	 * **Note:** A read-only component is not editable,
+	 * but still provides visual feedback upon user interaction.
+	 * @default false
+	 * @public
 	 */
 	@property({ type: Boolean })
-	preventPopoverOpen!: boolean;
+	readonly!: boolean;
 
 	/**
 	 * Indicates if the tokenizer should show all tokens or n more label instead
-	 * @private
+	 * @protected
 	 */
 	@property({ type: Boolean })
 	expanded!: boolean;
 
-	@property({ type: Object })
-	morePopoverOpener!: Tokenizer;
+	/**
+	 * Indicates if the nMore popover is open
+	 * @protected
+	 * @default false
+	 */
+	@property({ type: Boolean })
+	open!: boolean;
```

Comment:
Did we consider if this is needed for the applications to perform all desired actions for a fully functional tokenizer?

Reply:
Discussed internally that we will restrict the API until someone requires it.